### DISCUSSION TITLE
initcontainer misconfiguration check

### DISCUSSION THREAD

**Discussion by landbison1** [kubescape/kubescape#1734] [Q&A]
Thank you for your response the other day. 
I'm curious, kubescape checks securitycontext whenever it can, but  it doesn't check in initcontainer. initcontainer can do almost all container settings like privileged, allowPrivilegeEscalation, hostport, etc. in initcontainer, but it seems that initcontainer is less dangerous for security because of its short lifetime, so I want to know why there is no check for initcontainer.

**Comment by landbison1** [comment]
In particular, initcontainer can now have a long lifetime. In Kubernetes 1.29, we made it possible to use sidecar containers as initcontainer. https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/

**Reply by matthyx** [reply]
yes, as the owner of the KEP and co-host of the working group I know ;-)
I am not aware of any security framework considering these... but we can for sure add our own checks.
Do you mind opening an issue in https://github.com/kubescape/regolibrary/issues and/or https://github.com/kubescape/cel-admission-library ?
cc @slashben
